hazelcast/src/main/java/com/hazelcast/cluster/impl/ClusterClockImpl.java

```diff
@@ -46,4 +47,16 @@ public void setMasterTime(long masterTime) {
     public long getClusterTimeDiff() {
         return (clusterTimeDiff == Long.MAX_VALUE) ? 0 : clusterTimeDiff;
     }
+
+    @Override
+    public long getClusterStartTime() {
```

Comment:
Right now we do not need to handle these cases. If the user sets system clock to a previous time of the clusterStartTime, upTime value will be negative. However this case is not even handled by the JVM itself. if you try RuntimeMXBean's uptime() method, while setting system time back and forth, you will see it is not monotonic too. Also if the nodes clocks are not sync, since all the nodes will send its uptime values to call home server, we can group nodes with unique clusterId, try to calculate uptime with the other data provided in call home. Basically, you are right these problems are perfectly valid, but this I think this implementation will be enough for now.